theme/static/js/hs_file_browser.js

```diff
@@ -1847,18 +1847,26 @@ function onUploadSuccess(file, response) {
 
 $(document).ready(function () {
     // Download All method
-    $("#btn-download-all, #download-bag-btn").click(function (event) {
-        const btnDownloadAll = $("#btn-download-all");
-        const icon = $('#btn-download-all > span:first-child');
+    $("#btn-download-all, #btn-download-all-top-right, #download-bag-btn").click(function (event) {
+        let currentBtnDownloadAll = "btn-download-all";
+        if(event.currentTarget.id === "btn-download-all" || event.currentTarget.id === "btn-download-all-top-right") {
+            currentBtnDownloadAll = event.currentTarget.id;
+        }
+
+        const btnDownloadAll = $(`#${currentBtnDownloadAll}`);
+        const icon = $(`#${currentBtnDownloadAll} > span:first-child`);
         const initialClass = icon.attr("class");
         const dataAgreeRequired = btnDownloadAll.attr("data-toggle") === "modal";
 
         if (event.currentTarget.id === "download-bag-btn" || !dataAgreeRequired) {
             icon.attr("class", "fa fa-spinner fa-pulse fa-lg download-spinner");
+            if(event.currentTarget.id === "btn-download-all-top-right") {
+                icon.addClass("download-all-spinner-position"); 
+            }
             btnDownloadAll.css("cursor", "wait");
         }
         
-        if (dataAgreeRequired && event.currentTarget.id !== "download-bag-btn") {
+        if (dataAgreeRequired && ["download-bag-btn","btn-download-all-top-right"].indexOf(event.currentTarget.id) === -1) {
```

Comment:
There is a bug here.
In the case that a license agreement is required on the resource, when `#btn-download-all-top-right` is clicked, the download begins even before the user "accepts the terms for this licence":
https://github.com/user-attachments/assets/5b09ffd0-eb64-405d-8aef-084b23ae12f1